src/app/clusters/power-source-server/power-source-server.cpp

```diff
@@ -34,10 +35,55 @@ using namespace app::Clusters::PowerSource::Attributes;
 
 namespace {
 
+struct PowerSourceClusterInfo
+{
+    PowerSourceClusterInfo() : mClusterEndpoint(kInvalidEndpointId) {}
+    explicit PowerSourceClusterInfo(EndpointId powerClusterEndpointId) : mClusterEndpoint(powerClusterEndpointId) {}
+    void Shutdown()
+    {
+        mBuf.Free();
```

Comment:
This should be called in destructor to avoid false positives on leak detects.

Reply:
Unfortunately, this is a gotcha with some of the platforms, which shut down the memory subsystem before object destruction.

Reply:
followup issue for shutdown handling here - https://github.com/project-chip/connectedhomeip/issues/28457. The shutdown process (for all clusters actually) is not well defined and some of the platforms need to adjust the shutdown ordering, so I'd like to handle it in another PR.